core/src/io/anuke/mindustry/entities/effect/Lightning.java

```diff
@@ -65,6 +66,11 @@ public static void createLighting(int seed, Team team, Color color, float damage
             Bullet.create(Bullets.damageLightning, l, team, x, y, 0f, 1f, 1f, dmg);
             l.lines.add(new Vector2(x + Mathf.range(3f), y + Mathf.range(3f)));
 
+            if(hitsInsulator(l)) {
+                l.lines.remove(l.lines.size -1);
```

Comment:
This seems to work, but:

![image](https://user-images.githubusercontent.com/10100000/67637764-27614a80-f8b4-11e9-9c1d-fd983ff15c5e.png)

The lightning never really hits the wall, it just stops early. A workaround may involve the following:

```java
//outside the for-loop:
boolean[] hit = {false};

//inside loop
if(l.lines.size > 1){
    hit[0] = false;
    Position from = l.lines.get(l.lines.size - 2);
    Position to   = l.lines.get(l.lines.size - 1);
    world.raycastEach(x, y, x2, y2, (wx, wy) -> {

        Tile tile = world.ltile(wx, wy);
            if(tile != null && tile.block().insulated){
                hit[0] = true;
                //snap it instead of removing
                l.lines.get(l.lines.size -1).set(wx * tilesize, y * tilesize);
                return true;
            }
            return false;
    };
    if(hit[0]) break;
}
```

This would effectively snap the last lightning line to the block, making it look like it connected to something instead of stopping just short of the wall.

Also, this is more of a problem with my code, but
`Array<Position> lines` should probably be changed to `Array<Vector2> lines`. The `Position` type had some other purpose that isn't needed here anymore.

Reply:
> The lightning never really hits the wall, it just stops early. A workaround may involve the following:

+ you knew it stopped early right? 😅 

> I think this is the best option. There doesn't seem to be anything wrong with it besides the fact that it stops too early, yet continues to damage blocks in front. I'll look into implementing something like this.

Reply:
Well, yeah, I'm just saying it could look better with this method.

Reply:
Allrighty, i'll try to implement it later tonight 👍

Reply:
Had to rename hit (since it was taken), and replaced the xy's with variables i could find 😗